src/collection.js

```diff
@@ -494,6 +494,23 @@ class Collection {
     return this
   }
 
+  /**
+   * Returns median of the current collection
+   *
+   * @param {Function} comparator
+   *
+   * @returns {Collection}
+   */
+  async median () {
+    const arr = Array.from(this.items.sort())

```

Comment:
Can you please use the internal `.sort()` method?

Reply:
using internal sort will sort the current collection items. According to me that's not correct.
Because user is calculating the median  of the collection it should not sort the current collection.

Reply:
Let's take an example and see if we're talking about the same thing :)

Let's say you have the following collection pipeline:

```js
await Collect([1, 2, 3, 4, 10])
	.map(item => item * 2) // [2, 4, 6, 8, 20]
    .filter(item => item > 5) // [6, 8, 20]
    .median()

// expected result: 8
```
Do you agree with `8` as the expected result? Or do you think the result should be `3`?

Reply:
Yes expected result is correct i.e 8. 
Let's say you have an array [1,3,4,5]
`
await Collect([1,6,5]).median() // expect result : 6
`
but if we use the internal sort function first it will change items to [1,5,6] and then calculate median.
My method makes a copy of current items and perform operations on that not the original one.
Please suggest which implementation do you prefer

Reply:
The expected result of your example is `5`, not `6` because the items will be sorted before picking the median. Well, at least that's what I think the `median` does :)

```js
await Collect([1,6,5]).median()
// expected result: 5
```

The internal sort method also makes a copy of the items.

Reply:
I am very sorry expected result is 5.  😁 
Ok i will make this change

Reply:
Alright, no worries :smiley:

Reply:
Changes Done. Please Review